src/main/java/uk/gov/gchq/koryphe/impl/binaryoperator/StringDeduplicateConcat.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.koryphe.impl.binaryoperator;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import uk.gov.gchq.koryphe.binaryoperator.KorypheBinaryOperator;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * An <code>StringDeduplicateConcat</code> is a {@link KorypheBinaryOperator} that takes in
+ * {@link String}s, potentially with separators, and concatenates them together, omitting
+ * any duplicate substrings.
+ * The default separator is a comma, you can set a custom separator
+ * using setSeparator(String).
+ */
+public class StringDeduplicateConcat extends KorypheBinaryOperator<String> {
+
+    private static final String DEFAULT_SEPARATOR = ",";
+    private String separator = DEFAULT_SEPARATOR;
+
+    @Override
+    protected String _apply(final String a, final String b) {
+        final Set<String> set = new LinkedHashSet<>();
+
+        set.addAll(Arrays.asList(a.split(separator)));
+        set.addAll(Arrays.asList(b.split(separator)));
+
+        return set.stream().collect(Collectors.joining(separator));
```

Comment:
This is fine, I've normally just done StringUtils.join(set, ","); It would be useful to do a quick performance test to make sure we chose the fastest solution.